iocore/cache/CacheWrite.cc

```diff
@@ -102,10 +102,12 @@ CacheVC::updateVector(int /* event ATS_UNUSED */, Event * /* e ATS_UNUSED */)
     }
     if (vec) {
       /* preserve fragment offset data from old info. This method is
-         called iff the update is a header only update so the fragment
+         called if the update is a header only update so the fragment
          data should remain valid.
       */
-      if (alternate_index >= 0) {
+      // If we are not in header only updating case. Don't copy fragments.
+      if (alternate_index >= 0 &&
+          ((total_len == 0 && alternate.get_frag_offset_count() == 0) && !(f.allow_empty_doc && this->vio.nbytes == 0))) {
```

Comment:
If `total_len` is zero, then how are there any fragments? The same for `get_frag_offset_count() == 0`.  What is being copied by `copy_frag_offsets_from`? AFAICT this disables copies fragments entirely.

Reply:
IIUC, `alternate.copy_frag_offsets_from(write_vector->get(alternate_index));` means copy fragments table from previous alternate. `alternate_index` is set by `alternate_index = get_alternate_index(write_vector, update_key);` which update_key means the alternate which we update.

So `total_len == 0` means we don't write anything in this writing.  `alternate.get_frag_offset_count()` means we don't have any fragments. In this case, it is pure the header update case. we need to copy fragment table from old alternate.

>If total_len is zero, then how are there any fragments? The same for get_frag_offset_count() == 0.

Yes, you are right. Actually I haven't checked  whether the fragment is zero in small doc case.

Reply:
kindly ping @SolidWallOfCode ..